framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -2370,6 +2370,197 @@ bool VulkanReplayConsumerBase::CheckCommandBufferInfoForFrameBoundary(
+                                         "pLibraries[1]=%p,pLibraries[2]=%p,pLibraries[3]=%p",
+                                         pipelineLibraryInfo->libraryCount,
+                                         pipelineLibraryInfo->pLibraries[0],
+                                         pipelineLibraryInfo->pLibraries[1],
+                                         pipelineLibraryInfo->pLibraries[2],
+                                         pipelineLibraryInfo->pLibraries[3]);
+                }
+                else
+                {
+                    for (size_t k = 0; k < fragmentOutputPipelines.size(); k++)
+                    {
+                        if (libraryLinkInfo->pDescriptorCounts[2] == k)
+                        {
+                            pipelines[2] = fragmentOutputPipelines[k];
+                            break;
+                        }
+                    }
+
+                    libraries.swap(pipelines);
+                    pTempLibraries->pLibraries = libraries.data();
+                    GFXRECON_LOG_WARNING("[replay]vkCreateGraphicsPipelines:link:libraryCount=%d,pLibraries[0]=%p,"
+                                         "pLibraries[1]=%p,pLibraries[2]=%p",
+                                         pipelineLibraryInfo->libraryCount,
+                                         pipelineLibraryInfo->pLibraries[0],
+                                         pipelineLibraryInfo->pLibraries[1],
+                                         pipelineLibraryInfo->pLibraries[2]);
+                }
+                pTempLibraries->pNext = nullptr;
+            }
+            else
+            {
+                GFXRECON_LOG_WARNING("[replay]vkCreateGraphicsPipelines:link:error. libraryLinkInfo == nullptr.");
+            }
+        }
+        else
+        {
+            GFXRECON_LOG_WARNING("[replay]vkCreateGraphicsPipelines:link:gpl off. libraryInfo == nullptr.");
+        }
+    }
+    else if (createInfo->flags == 0x0)
+    {
+        GFXRECON_LOG_WARNING("[replay]vkCreateGraphicsPipelines:link:gpl off. in_pCreateInfos->pNext == nullptr.");
+    }
+}
+
+void VulkanReplayConsumerBase::CompileGraphicsPipelines(const VkGraphicsPipelineCreateInfo* createInfo,
+                                                        VkPipeline*                         pipeline)
+{
+    if (createInfo->flags & VK_PIPELINE_CREATE_LIBRARY_BIT_KHR)
+    {
+        const VkPipelineCreationFeedbackCreateInfo* feedbackInfo =
+            (reinterpret_cast<const VkPipelineCreationFeedbackCreateInfo*>(createInfo->pNext));
+        const VkGraphicsPipelineLibraryCreateInfoEXT* libraryInfo =
+            (reinterpret_cast<const VkGraphicsPipelineLibraryCreateInfoEXT*>(feedbackInfo->pNext));
+        GFXRECON_LOG_WARNING(
+            "[replay]vkCreateGraphicsPipelines:compile:flags[0x%x], pPipelines=%p", libraryInfo->flags, *pipeline);
+        if (libraryInfo->flags == VK_GRAPHICS_PIPELINE_LIBRARY_VERTEX_INPUT_INTERFACE_BIT_EXT)
```

Comment:
Since these are `BIT` values and not an enumerant, this comparison and the ones below would need to be anded with the BIT to test first, to handle the (unlikely) case that a library contains multiple subsets.